add support for Radxa ROCK 3C, Radxa ZERO 3E and 3W

RadxaNaoki opened the pull request:
please read commit messages for the detail.

RadxaNaoki committed:
uboot-rockchip: renumber patches

renumber patches for upcoming new device support

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>

RadxaNaoki committed:
rockchip: add support for Radxa ROCK 3C

Radxa ROCK 3C is a high-performance, low-cost SBC[1] using the
Rockchip RK3566.

Hardware
--------
- Rockchip RK3566 SoC
- Quad A55 CPU
- Mali-G52-2EE GPU
- 1 TOPS @ INT8 NPU
- 1GB/2GB/4GB LPDDR4 RAM
- eMMC connector
- Micro SD Card slot
- NVMe SSD through the M.2 M Key connector(2230) or M.2 Extension
  board(2232/2260/2280)
- SATA through the Radxa Penta SATA HAT
- 1x Gigabit ethernet port(supports PoE with add-on PoE HAT)
- WiFi6/BT5.4 (not supported yet on OpenWrt)
- 1x USB 3.0 Type-A HOST port
- 2x USB 2.0 Type-A HOST ports
- 1x USB 2.0 Type-A OTG port
- 40 Pin GPIO header

[1] https://radxa.com/products/rock3/3c

Installation
------------
Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
internal eMMC using dd.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>

RadxaNaoki commented:
it seems I did something wrong in target/linux/rockchip/patches-6.6/047-*. I'll fix it...

maurerr commented on package/boot/uboot-rockchip/patches/000-backport-upstream-dts-sync.patch:
why sync other dts files except zero3e/w+rock3c?

maurerr commented:
cc @1715173329

RadxaNaoki commented:
> The first commit title should start with `uboot-rockchip`, otherwise LGTM.
> 
> EDIT: please also refresh uboot patches with `make package/uboot-rockchip/refresh V=s`

done

thanks!

RadxaNaoki commented:
conflict with rk3588 PRs...

RadxaNaoki committed:
rockchip: add support for Radxa ZERO 3E and 3W

Radxa ZERO 3E and 3W are light, compact and tiny SBC[1][2] using the
Rockchip RK3566.

Hardware
--------
- Rockchip RK3566 SoC
- Quad A55 CPU
- Mali-G52-2EE GPU
- 1 TOPS @ INT8 NPU
- 1GB/2GB/4GB/8G LPDDR4 RAM
- Optional 8GB/16GB/32GB/64GB eMMC (3E)
- Micro SD Card slot
- 1x Gigabit ethernet port (supports PoE with add-on PoE HAT) (3E)
- WiFi6/BT5.4 (3W) (not supported yet on OpenWrt)
- 1x USB 3.0 Type-C HOST port
- 1x USB 2.0 Type-C OTG port
- Optional 40 Pin GPIO header

[1] https://radxa.com/products/zeros/zero3e
[2] https://radxa.com/products/zeros/zero3w

Installation
------------
Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
internal eMMC using dd.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>

RadxaNaoki commented:
is there any issue to merge this PR?

maurerr commented:
we need one of the maintainers to find time - maybe @hauke or @PolynomialDivision

RadxaNaoki commented:
may I close this PR and migrate to https://github.com/openwrt/openwrt/pull/16185 ?

hauke commented:
The CI found a problem in this PR:
```

Run . .github/workflows/scripts/ci_helpers.sh
target/linux/rockchip/patches-6.6/053-v6.9-arm64-dts-rockchip-Add-support-for-NanoPi-R6S.patch
Kernel patches for rockchip/armv8 require refresh. (run 'make target/linux/refresh' and force push this pr)
You can also check the provided artifacts with the refreshed patch from this CI run.
Error: Process completed with exit code 1.
```

RadxaNaoki commented:
> The CI found a problem in this PR:
> 
> ```
> 
> Run . .github/workflows/scripts/ci_helpers.sh
> target/linux/rockchip/patches-6.6/053-v6.9-arm64-dts-rockchip-Add-support-for-NanoPi-R6S.patch
> Kernel patches for rockchip/armv8 require refresh. (run 'make target/linux/refresh' and force push this pr)
> You can also check the provided artifacts with the refreshed patch from this CI run.
> Error: Process completed with exit code 1.
> ```

I know. there was a patch for it. but I squashed it by request https://github.com/openwrt/openwrt/pull/16185#pullrequestreview-2256397821
